### PR TITLE
Oci project structure

### PR DESCRIPTION
**Pull request by ameydwivedi**
updated project to enable authorization. Tested with getting List Instances API.

@RajeshKalyankar , @manjoosh12  please review.

### CODE REVIEW

**Comment by ameydwivedi** [comment]
@patmante , please review

File: cs-oracle-cloud/src/main/java/io/cloudslang/content/oracle/entities/signature/Signature.java [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+        RequestSigner signer = new RequestSigner(apiKey, privateKey);
+
+        System.out.println(uri);
+        request = new HttpGet(uri);
+
+        signer.signRequest(request);
+        Header[] auth = request.getHeaders("Authorization");
+        Header[] date = request.getHeaders("date");
+        Header[] host = request.getHeaders("host");
+        System.out.println("Authorization Header: " + auth[0].toString().substring(15, auth[0].toString().length()));
```
Comment (patmante) [inline cs-oracle-cloud/src/main/java/io/cloudslang/content/oracle/entities/signature/Signature.java:77]: remove SOPs

File: cs-oracle-cloud/src/main/java/io/cloudslang/content/oracle/utils/Constants.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        public static final String PAGE_SIZE = "page[size]=";
+        public static final String SESSION_TIMEOUT = "20160";
+        public static final String SESSION_REMEMBER = "20160";
+        public static final String COLLABORATOR_AUTH_POLICY = "password";
+    }
+
+
+
+    public static class ListInstancesConstants {
+        public static final String LIST_INSTANCES_OPERATION_NAME = "List Organizations";
```
Comment (patmante) [inline cs-oracle-cloud/src/main/java/io/cloudslang/content/oracle/utils/Constants.java:68]:
```suggestion
        public static final String LIST_INSTANCES_OPERATION_NAME = "List Organizations";
``` 
Change the operation name or remove it

File: cs-oracle-cloud/src/main/java/io/cloudslang/content/oracle/utils/Descriptions.java [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.cloudslang.content.oracle.utils;
+
+public class Descriptions {
+    public static class Common {
+        public static final String TENANCY_OCID_DESC =  "Tenancy OCID of the OCI account.";
```
Comment (patmante) [inline cs-oracle-cloud/src/main/java/io/cloudslang/content/oracle/utils/Descriptions.java:20]: update the description properly

File: cs-oracle-cloud/src/main/java/io/cloudslang/content/oracle/utils/Descriptions.java [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+        public static final String SUCCESS_DESC = "The request was successfully executed.";
+        public static final String RETURN_RESULT_DESC = "If successful, returns the complete API response. In case of an error this output will contain the error message.";
+    }
+
+
+    public static class ListOAuthClient {
+        public static final String DOCUMENT_DESC = "The full API response in case of success.";
+        public static final String OAUTH_TOKEN_ID_DESCRIPTION = "The VCS Connection (OAuth Connection + Token) to use. " +
+                "This ID can be obtained from the oauth-tokens endpoint.";
+        public static final String LIST_OAUTH_CLIENT_DESC = "List An OAuth Client Id";
```
Comment (patmante) [inline cs-oracle-cloud/src/main/java/io/cloudslang/content/oracle/utils/Descriptions.java:95]: remove reference to terraform project

File: cs-oracle-cloud/src/main/java/io/cloudslang/content/oracle/utils/Outputs.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+import io.cloudslang.content.constants.OutputNames;
+
+public class Outputs extends OutputNames {
+    public static class CommonOutputs {
+        public static final String DOCUMENT = "document";
+
+    }
+
+    public static class ListOAuthClientOutputs {
```
Comment (patmante) [inline cs-oracle-cloud/src/main/java/io/cloudslang/content/oracle/utils/Outputs.java:26]: remove the reference to Terraform.

File: cs-oracle-cloud/src/main/java/io/cloudslang/content/oracle/utils/HttpUtils.java [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+                public PasswordAuthentication getPasswordAuthentication() {
+                    return new PasswordAuthentication(proxyUser, proxyPassword.toCharArray());
+                }
+            });
+        }
+        return new Proxy(HTTP, InetSocketAddress.createUnresolved(proxyHost, proxyPort));
+    }
+
+    public static void setProxy(@org.jetbrains.annotations.NotNull final HttpClientInputs httpClientInputs,
+                                @org.jetbrains.annotations.NotNull final String proxyHost,
```
Comment (patmante) [inline cs-oracle-cloud/src/main/java/io/cloudslang/content/oracle/utils/HttpUtils.java:52]: add secure properties

File: cs-oracle-cloud/src/main/java/io/cloudslang/content/oracle/entities/signature/Signature.java [hunk 7]
```diff
@@ -0,0 +1,286 @@
...
+import java.io.InputStream;
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.security.Key;
+import java.security.PrivateKey;
+import java.security.spec.InvalidKeySpecException;
+import java.text.SimpleDateFormat;
+import java.util.*;
```
Comment (patmante) [inline cs-oracle-cloud/src/main/java/io/cloudslang/content/oracle/entities/signature/Signature.java:41]: if possible try to import required classes

File: cs-oracle-cloud/src/main/java/io/cloudslang/content/oracle/utils/Constants.java [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+        public static final String DEFAULT_JAVA_KEYSTORE = System.getProperty("java.home") + "/lib/security/cacerts";
+        public static final String CHANGEIT = "changeit";
+        public static final String ZERO = "0";
+        public static final String CONNECT_TIMEOUT_CONST = "10000";
+        public static final String UTF8 = "UTF-8";
+        public static final String CONNECTIONS_MAX_PER_ROUTE_CONST = "2";
+        public static final String CONNECTIONS_MAX_TOTAL_CONST = "20";
+        public static final String AUTHORIZATION = "Authorization:";
+        public static final String BEARER = "Bearer ";
+        public static final String OCI_HOST = "iaas.us-ashburn-1.oraclecloud.com";
```
Comment (patmante) [inline cs-oracle-cloud/src/main/java/io/cloudslang/content/oracle/utils/Constants.java:45]: make it dynamic not constant host.

File: cs-oracle-cloud/src/main/java/io/cloudslang/content/oracle/utils/Constants.java [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+        public static final String LIST_INSTANCES = "/instances";
+        public static final String COMPARTMENT_ID = "compartmentId=";
+        public static final String PATH_SEPARATOR = "/";
+        public static final String AND = "&";
+        public static final String QUERY = "?";
+        public static final String ID = "id";
+        public static final String HTTPS = "https";
+        public static final String STATUS_CODE = "statusCode";
+        public static final String APPLICATION_VND_API_JSON = "application/vnd.api+json";
```
Comment (patmante) [inline cs-oracle-cloud/src/main/java/io/cloudslang/content/oracle/utils/Constants.java:54]: remove this or update it if you are using

File: cs-oracle-cloud/src/main/java/io/cloudslang/content/oracle/utils/Descriptions.java [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.cloudslang.content.oracle.utils;
+
+public class Descriptions {
+    public static class Common {
+        public static final String TENANCY_OCID_DESC =  "Oracle creates a tenancy for your company, which is a secure and isolated partition where you can create, organize, and administer your cloud resources. This is ID of the tenancy.";
```
Comment (patmante) [inline cs-oracle-cloud/src/main/java/io/cloudslang/content/oracle/utils/Descriptions.java:20]: wrap the description into multiple lines

File: cs-oracle-cloud/src/main/java/io/cloudslang/content/oracle/utils/Descriptions.java [hunk 11]
```diff
@@ -0,0 +1,116 @@
...
+
+
+    public static class ListOAuthClient {
+        public static final String DOCUMENT_DESC = "The full API response in case of success.";
+
+    }
+
+    public static class ListInstances {
+        public static final String LIST_INSTANCES_OPERATION_DESC = "List of instances present";
+        public static final String INSTANCE_LIST_DESC = "List of all instances";
```
Comment (patmante) [inline cs-oracle-cloud/src/main/java/io/cloudslang/content/oracle/utils/Descriptions.java:98]: is this used in code or duplicate?

File: cs-oracle-cloud/src/main/java/io/cloudslang/content/oracle/utils/Outputs.java [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+
+public class Outputs extends OutputNames {
+    public static class CommonOutputs {
+        public static final String DOCUMENT = "document";
+
+    }
+
+
+    public static class ListInstancesOutputs {
+        public static final String INSTANCE_LIST = "instance_list";
```
Comment (patmante) [inline cs-oracle-cloud/src/main/java/io/cloudslang/content/oracle/utils/Outputs.java:28]: output is in this format or "instanceList"?

File: cs-oracle-cloud/src/main/java/io/cloudslang/content/oracle/entities/inputs/OCICommonInputs.java [hunk 13]
```diff
@@ -0,0 +1,103 @@
...
+        this.privateKeyFilename = privateKeyFilename;
+    }
+
+    @NotNull
+    public static OCICommonInputs.OCICommonInputsBuilder builder() {
+        return new OCICommonInputs.OCICommonInputsBuilder();
+    }
+
+    @NotNull
+    public String getTenancyOcId() {
```
[the remark this answers is not in the document]
Comment (ameydwivedi) [reply cs-oracle-cloud/src/main/java/io/cloudslang/content/oracle/entities/inputs/OCICommonInputs.java:44]: updated

File: cs-oracle-cloud/src/main/java/io/cloudslang/content/oracle/utils/Constants.java [hunk 14]
```diff
@@ -0,0 +1,89 @@
...
+        public static final String OCI_HOST = "iaas.us-ashburn-1.oraclecloud.com";
+        public static final String LIST_INSTANCES = "/instances";
+        public static final String COMPARTMENT_ID = "compartmentId=";
+        public static final String PATH_SEPARATOR = "/";
+        public static final String AND = "&";
+        public static final String QUERY = "?";
+        public static final String ID = "id";
+        public static final String HTTPS = "https";
+        public static final String STATUS_CODE = "statusCode";
+        public static final String APPLICATION_VND_API_JSON = "application/json";
```
Comment (patmante) [inline cs-oracle-cloud/src/main/java/io/cloudslang/content/oracle/utils/Constants.java:54]: change the APPLICATION_VND_API_JSON to APPLICATION_JSON

Comment (ameydwivedi) [reply cs-oracle-cloud/src/main/java/io/cloudslang/content/oracle/utils/Constants.java]: updated

**Pull request merged**